src/main/java/prefs/Preferences.java

```diff
@@ -11,122 +17,262 @@
 /**
  * Represents persistent user configuration. Maps to a file on disk.
  *
+ * The Preferences class is the source of all configuration and preferences information. All preferences and config
+ * settings can be set in this class.
+ *
+ * Use either Preferences.create() or Preferences.load() to obtain a Preferences instance.
+ *
  * Overrides PMD's recommendation that this class should be final.
  * It cannot be as we need to mock it.
+ * 
  */
 public class Preferences { // NOPMD
+
+    private static final Logger logger = LogManager.getLogger(Preferences.class.getName());
+
     public static final String DIRECTORY = "settings";
 
-    // Standard config filenames used for application and testing
-    public static final String GLOBAL_CONFIG_FILE = "global.json";
-    public static final String TEST_CONFIG_FILE = "test.json";
+    public static final String SESSION_CONFIG_FILENAME = "global.json";
+    public static final String USER_CONFIG_FILENAME = "user.json";
 
-    private final ConfigFileHandler fileHandler;
+    private static final String DEFAULT_FILE_CONTENTS = "{}";
 
-    public GlobalConfig global;
+    private final String configDirectory;
+    private final String sessionConfigFileName;
+    private final String userConfigFileName;
 
-    private Preferences(String configFileName, boolean createUnconditionally) {
-        this.fileHandler = new ConfigFileHandler(DIRECTORY, configFileName);
+    private final SessionConfig sessionConfig;
+    private final UserConfig userConfig;
 
-        if (createUnconditionally) {
-            initGlobalConfig();
+    /**
+     * Initialises a Preferences instance that contain the configurations in the config files specified
+     * @param configDirectory The directory that the config files are held in.
+     * @param sessionConfigFileName The file name of the session config file which stores session config values.
+     * @param userConfigFileName The file name of the user config file which stores user config values.
```

Comment:
For here and below, maybe a more concise description can be used. Something like `the file name of the serialized <or jsonified> UserConfig`. Or maybe `the file name of UserConfig object that's stored as file`.